pkg/upgrade/hub/hub.go

```diff
@@ -86,7 +89,11 @@ func (h *HubClient) Info(repo, name string) (info FList, err error) {
 	}
 
 	u.Path = filepath.Join("api", "flist", repo, name, "light")
-	response, err := http.Get(u.String())
+	cl := &http.Client{
```

Comment:
the HubClient can obviously also use a client attribute instead of re-creating the client. This way it can use the connection reuse features of http2 and saves on memory